apps/transport/lib/irve/database_importer.ex

```diff
@@ -0,0 +1,101 @@
+defmodule Transport.IRVE.DatabaseImporter do
+  @moduledoc """
+  A module to import IRVE data files and their PDC into the database.
+  It assumes you have a path to a valid IRVE data file.
+  """
+
+  import Ecto.Query
+
+  def write_to_db(file_path, dataset_datagouv_id, resource_datagouv_id) do
+    {:ok, content} = File.read(file_path)
+
+    rows =
+      content |> Transport.IRVE.Processing.read_as_data_frame() |> Explorer.DataFrame.to_rows()
+
+    #  Note for review: may chose another way to get a checksum
+    # See Transport.S3.AggregatesUploader sha256! private function
+    checksum = :crypto.hash(:sha256, content) |> Base.encode16(case: :lower)
+
+    previous_version = get_previous_file(dataset_datagouv_id, resource_datagouv_id)
+
+    case previous_version do
+      #  If previous version exists and checksum is the same, do nothing.
+      %DB.IRVEValidFile{checksum: ^checksum} ->
+        :no_change
+
+      # If there is no previous version, insert one. Then insert all PDCs.
+      nil ->
+        DB.Repo.transaction(fn ->
+          {:ok, %DB.IRVEValidFile{id: file_id}} = write_new_file(dataset_datagouv_id, resource_datagouv_id, checksum)
+          write_pdcs(rows, file_id)
+        end)
+
+      # If previous version exists but checksum is different, update it, then delete all PDCs and reinsert them.
+      %DB.IRVEValidFile{} ->
+        DB.Repo.transaction(fn ->
+          {:ok, %DB.IRVEValidFile{id: file_id}} = update_file(previous_version, checksum)
+          # Note for review: we have to delete previous PDCS before as there is no version identifier
```

Comment:
@thbar à ton avis faut-il garder une colonne hash sur la table des PDC pour savoir si on se réfère à la bonne version ?

Reply:
@vdegove en l'état, `irve_valid_pdc` belongs to `irve_valid_pdc`, qui porte le checksum du fichier. Si je te comprends bien ça serait redondant (on aurait déjà l'info), correct ?